### PR TITLE
Configuration for NFVPE plugins

### PR DESCRIPTION
**Pull request by paramite**
#### Pull Request (PR) description
This patch adds support for NFVPE plugins, namely: sysevent, procevent and connectivity. 
Those plugins are not merged yet in collectd itself, but once this (hopefully) happens, 
we will have configuration patch ready for merge. 

Collectd patches:
https://github.com/collectd/collectd/pull/2624
https://github.com/collectd/collectd/pull/2623
https://github.com/collectd/collectd/pull/2622

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

File: manifests/plugin/connectivity.pp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#  Defaults to $collectd::manage_package
+#
+# [*interfaces*]
+#  Array of interface(s) to monitor connect to. Empty arrayf means all interfaces
+#  Defaults to []
+#
+class collectd::plugin::connectivity (
+  Enum['present', 'absent'] $ensure = 'present',
+  Boolean $manage_package           = $collectd::manage_package,
+  Array[String] $interfaces         = [],
```
Comment (bastelfreak) [inline manifests/plugin/connectivity.pp]: can you update this to `Array[String[1]]`? That enforces a minimal string length of one.

File: manifests/plugin/sysevent.pp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#  Maximum number of rsyslog events that can be stored in plugin's ring buffer.
+#  Once an event has been read, its location becomes available for storing
+#  a new event.
+#  Defaults to undef
+#
+class collectd::plugin::sysevent (
+  Enum['present', 'absent'] $ensure  = 'present',
+  Boolean $manage_package            = $collectd::manage_package,
+  Stdlib::Host $listen_host          = '127.0.0.1',
+  Integer $listen_port               = 6666,
```
Comment (bastelfreak) [inline manifests/plugin/sysevent.pp]: Can you use this type please? https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/port.pp. We then need to make sure that stdlib 4.25.0 is the lowest supported version in our metadata.json

File: manifests/plugin/procevent.pp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#  Once an event has been read, its location becomes available for storing
+#  a new event.
+#  Defaults to undef
+#
+class collectd::plugin::procevent (
+  Enum['present', 'absent'] $ensure  = 'present',
+  Boolean $manage_package            = $collectd::manage_package,
+  Optional[String] $process          = undef,
+  Optional[String] $regex_process    = undef,
+  Optional[Integer] $buffer_length   = undef,
```
Comment (bastelfreak) [inline manifests/plugin/procevent.pp]: You can also enforce that this has to be a positive integer: https://puppet.com/docs/puppet/5.5/lang_data_number.html#the-integer-data-type

File: manifests/plugin/connectivity.pp (hunk 1 above)
Comment (paramite) [reply manifests/plugin/connectivity.pp]: Will do.

File: manifests/plugin/sysevent.pp (hunk 2 above)
Comment (paramite) [reply manifests/plugin/sysevent.pp]: Will do.

File: manifests/plugin/procevent.pp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# [*buffer_length*]
+#  Maximum number of rsyslog events that can be stored in plugin's ring buffer.
+#  Once an event has been read, its location becomes available for storing
+#  a new event.
+#  Defaults to undef
+#
+class collectd::plugin::procevent (
+  Enum['present', 'absent'] $ensure            = 'present',
+  Boolean $manage_package                      = $collectd::manage_package,
+  Optional[String] $process                    = undef,
```
Comment (bastelfreak) [inline manifests/plugin/procevent.pp]: can you use  `Optional[String[1]]` here?

File: manifests/plugin/procevent.pp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  Optional[String] $regex_process              = undef,
```
Comment (bastelfreak) [inline manifests/plugin/procevent.pp]: can you use  `Optional[String[1]]` here?

Comment (paramite) [reply manifests/plugin/procevent.pp]: Sure

File: manifests/plugin/procevent.pp (hunk 4 above)
Comment (paramite) [reply manifests/plugin/procevent.pp]: Sure.

**Comment by bastelfreak** [comment]
Thanks!

**Pull request merged**